multi-transfer-esdt/src/lib.rs

```diff
@@ -56,6 +57,18 @@ pub trait MultiTransferEsdt:
         let safe_address = self.esdt_safe_contract_address().get();
 
         for eth_tx in transfers {
+            // Check if token has TransferRole
+            if self
+                .blockchain()
+                .get_esdt_local_roles(&eth_tx.token_id)
+                .has_role(&EsdtLocalRole::Transfer)
+            {
+                let refund_tx = self.convert_to_refund_tx(eth_tx);
```

Comment:
This is a discussion a little more complex than the proposed solution. I would postpone it to the next release and go deep into all scenarios